src/Adyen/Util/Util.php

```diff
@@ -78,14 +78,69 @@ public static function formatAmount($amount, $currency)
         return (int)number_format($amount, $format, '', '');
     }
 
-    public static function isValidHmac($params, $key)
+    public static function calculateNotificationHMAC($params, $hmacKey)
+    {
+        // validate if hmacKey is provided
+        if (empty($hmacKey)) {
+            throw new \Adyen\AdyenException("You did not provide a HMAC key");
+        }
+
+        // validate if hmacKey contains only hexadecimal chars to be packed with H*
+        if (!ctype_xdigit($hmacKey)) {
+            throw new \Adyen\AdyenException("Invalid HMAC key: $hmacKey");
+        }
+
+        if (empty($params)) {
+            throw new \Adyen\AdyenException("You did not provide any parameters");
+        }
+
+        $dataToSign = self::getNotificationDataToSign($params);
+
+        // base64-encode the binary result of the HMAC computation
+        $merchantSig = base64_encode(hash_hmac('sha256', $dataToSign, pack("H*", $hmacKey), true));
+        return $merchantSig;
+    }
+
+    public static function getNotificationDataToSign($params)
+    {
+        $pspReference = (!empty($params['pspReference'])) ? $params['pspReference'] : "";
+        $originalReference = (!empty($params['originalReference'])) ? $params['originalReference'] : "";
+        $merchantAccountCode = (!empty($params['merchantAccountCode'])) ? $params['merchantAccountCode'] : "";
+        $merchantReference = (!empty($params['merchantReference'])) ? $params['merchantReference'] : "";
+        $value = (!empty($params['amount']['value'])) ? $params['amount']['value'] : "";
+        $currency = (!empty($params['amount']['currency'])) ? $params['amount']['currency'] : "";
+        $eventCode = (!empty($params['eventCode'])) ? $params['eventCode'] : "";
+        $success = (!empty($params['success'])) ? $params['success'] : "";
+
+        $dataToSign = array(
+            $pspReference,
+            $originalReference,
+            $merchantAccountCode,
+            $merchantReference,
+            $value,
+            $currency,
+            $eventCode,
+            $success
+        );
+
+        // The character escape function
+        $escapeval = function ($val) {
+            return str_replace(':', '\\:', str_replace('\\', '\\\\', $val));
```

Comment:
We can replace this with `preg_replace('/(:|\\)/g', '\\$1', $val)` or something like this. This way we remove the nested function calls, but I understand that it adds regular expressions (and a not very readable one).

Another option would be to call them in two separate lines instead.